packages/rest/src/parser.ts

```diff
@@ -95,34 +93,59 @@ function buildOperationArguments(
 ): OperationArgs {
   const args: OperationArgs = [];
 
+  let requestBodyIndex: number = -1;
+  if (hasArgumentsFromBody(operationSpec)) {
+    const i = (<RequestBodyObject>operationSpec.requestBody)[
+      REQUEST_BODY_INDEX
+    ];
+    requestBodyIndex = i ? i : 0;
+  }
+
+  const paramArgs: OperationArgs = [];
   for (const paramSpec of operationSpec.parameters || []) {
     if ('$ref' in paramSpec) {
       // TODO(bajtos) implement $ref parameters
       // See https://github.com/strongloop/loopback-next/issues/435
       throw new Error('$ref parameters are not supported yet.');
     }
+
     const spec = paramSpec as ParameterObject;
     switch (spec.in) {
       case 'query':
-        args.push(request.query[spec.name]);
+        paramArgs.push(request.query[spec.name]);
         break;
       case 'path':
-        args.push(pathParams[spec.name]);
+        paramArgs.push(pathParams[spec.name]);
         break;
       case 'header':
-        args.push(request.headers[spec.name.toLowerCase()]);
+        paramArgs.push(request.headers[spec.name.toLowerCase()]);
         break;
-      case 'formData':
-        args.push(body ? body[spec.name] : undefined);
-        break;
-      case 'body':
-        args.push(body);
+      case 'cookie':
+        // For details about cookie format, please check:
+        // https://www.npmjs.com/package/cookie
+        let cookies = cookieParser(request.headers.cookie || '');
```

Comment:
Cookie support is not so straightforward. It's a recommended practice to encrypt cookie values using a server-configured secret. We will need a way how to tell `@loopback/rest` what is this secret, otherwise it won't be able to decode cookie values.

Let's leave "in: cookie" parameters out of scope of this pull request and also the MVP scope. Action items:
 1. Open a new issue in loopback-next - a feature request to support "in: cookie" parameters
 2. Throw a helpful error here, explain that cookie parameters are not supported yet and point people to the GH issue tracking this feature.

Reply:
👍 sounds good! 
issue created: https://github.com/strongloop/loopback-next/issues/997

Will add the error msg for it in subsequent PR.